### DISCUSSION TITLE
Run from source (not git)

### DISCUSSION THREAD

**Discussion by BobSquarePants** [cdgriffith/FastFlix#619] [Q&A]
Hi,

I've downloaded the source of the version [5.8.1](https://github.com/cdgriffith/FastFlix/releases/tag/5.8.1) 
https://github.com/cdgriffith/FastFlix/archive/refs/tags/5.8.1.tar.gz

 I've unpack them, but I don't see how launch it.

I've tried 
```bash
cd fastflix
python3 flix.py
```
> ModuleNotFoundError: No module named 'packaging'

Thanks.

( out of topic, awesome software !! )

**Comment by cdgriffith** [comment]
Hello @BobSquarePants, thank you ! 

Should be the same commands as if running from git, just without the git part! 

## Run from source

You will need Python 3.12 ~and git~

**Setup**
```
cd FastFlix
python3.12 -m pip install --upgrade pip
python3.12 -m venv venv
. ./venv/bin/activate       # venv\Scripts\activate.bat or venv\Scripts\activate.ps1 on windows   
pip install .
```

**Running**

Linux

```
venv/bin/python -m fastflix
```

Windows

```
venv\Scripts\python.exe -m fastflix
```

**Comment by BobSquarePants** [comment]
Thanks @cdgriffith 

I've tried your commands, but didn't works too :/

`python3.11 -m pip install --upgrade pip`
> /usr/bin/python3.11: No module named pip

Then I tried

```bash
python3.11 -m venv .
source ./bin/activate #succeeded
pip install --upgrade pip #succeeded
pip install . # Warning require www connection
./bin/python -m fastflix
```

> Traceback (most recent call last):
  File "/home/\<bob>/Desktop/FastFlix/fastflix/entry.py", line 8, in <module>
    import coloredlogs
ModuleNotFoundError: No module named 'coloredlogs'
Could not load FastFlix properly!

**Reply by cdgriffith** [reply]
That's weird as it should be installed as part of the `install .`  https://github.com/cdgriffith/FastFlix/blob/master/pyproject.toml#L30C6-L30C17

Did you try manually installing that `pip install coloredlogs`? Were there any errors during pip install or just warnings?
